extract_unique_sensors.py

```diff
@@ -15,6 +15,7 @@
 for sensor in candidate_lines:
     # Typical output from "sensor" variable:
     # kat.sensors.anc_api_version   kat.sensors.anc_ganglia_api_version   kat.sensors.anc_ganglia_kat_monctl_system_os_name
-    sensor = sensor.split('.')[-1].strip("\n")
+    sensor = sensor.split('.')[-1] # to split a string using a 'dot'character.
+    sensor.strip("\n") # to remove newlines in a string
```

Comment:
This has no effect, you are merely stripping of `\n` and not overwriting `sensor` variable.

This is what you should do instead:
`sensor = sensor.strip("\n")`